odl/contrib/README.md

```diff
@@ -0,0 +1,98 @@
+
+- outside the main ODL library,
+- experimental and untested,
+- still subject to major breaking changes or
+- "not quite there yet" in terms of documentation, testing or completeness.
+
+The general idea is that you get your code quickly merged without *having to* meet the quality and consistency standards of main ODL that we ensure through extended code reviews. The reasons why we still want your contributed code to be part of the main ODL *repository* are (1) that ODL is intended to follow a ["batteries included"](https://www.python.org/dev/peps/pep-0206/#batteries-included-philosophy) philosophy, and (2) to give you as a contributor maximum visibility.
+
+**Important:** Responsibility for code that you contribute will remain with you. This means in particular that issues on the tracker about your module will be assigned to you. Of course, others will try to help if necessary.
+
+**Note:** If you consider your code to be useful for the main library, please consider filing a pull request anyway. We try to be as helpful as possible in our reviews and make suggestions, so that getting the code ready for merging doesn't take too long.
+Note also that contributed modules that attract significant attention and user adoption will be considered for core ODL.
+
+Here are some guidelines for contributors regarding the scope and the rules of this platform.
+
+
+## Scope
+
+**Yes:**
+
+- Generic code that would fit into main ODL but is still experimental,
+- Special-purpose code that is not generic enough to be a fit for main ODL (e.g. [figures of merit](https://github.com/odlgroup/odl/pull/1018) or application-specific code),
+- Extended, more complex examples than [the ones in main ODL](https://github.com/odlgroup/odl/tree/master/examples).
+
+**No:**
+
+- Code for specialized algorithms written specifically for your latest paper,
+- Large files like images, datasets etc.,
+- Huge amounts of code with questionable usefulness,
+- Code under licenses other than MPL-2.0.
+
+For code of the type "your latest paper", we recommend using a separate Github repository like [this one](https://github.com/kohr-h/variable_lp_paper) and making a pull request to add your project to the "Projects using ODL" section of the main README.
+
+
+## Procedure
+
+Since you do not have write access to the target location in the ODL repository, do the initial work on the code on [your personal fork](https://odl.readthedocs.io/dev/gitwash/forking_hell.html#forking). Within this fork, add your contribution to the `odl/contrib` directory. When the code is ready, the steps are as follows:
+
+- Submit a pull request. If you would like an in-depth review, please indicate so in writing or by choosing a reviewer on the Github pull request page.
+- We will check the pull request as a whole and (optionally) the code. Note that your tests must pass and the code must adhere to [PEP8](https://www.python.org/dev/peps/pep-0008/) for Travis CI to go green.
```

Comment:
~~for Travis CI to go green~~

since this is part of the automatic tests run on all code.